envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/permissions/IncomingPermissionsParamsInEndpointPathTest.kt

```diff
@@ -0,0 +1,251 @@
+package pl.allegro.tech.servicemesh.envoycontrol.permissions
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.testcontainers.junit.jupiter.Container
+import pl.allegro.tech.servicemesh.envoycontrol.assertions.hasNoRBACDenials
+import pl.allegro.tech.servicemesh.envoycontrol.assertions.hasOneAccessDenialWithActionBlock
+import pl.allegro.tech.servicemesh.envoycontrol.assertions.isRbacAccessLog
+import pl.allegro.tech.servicemesh.envoycontrol.config.Echo1EnvoyAuthConfig
+import pl.allegro.tech.servicemesh.envoycontrol.config.Echo2EnvoyAuthConfig
+import pl.allegro.tech.servicemesh.envoycontrol.config.EnvoyControlTestConfiguration
+import pl.allegro.tech.servicemesh.envoycontrol.config.containers.ToxiproxyContainer
+import pl.allegro.tech.servicemesh.envoycontrol.config.envoy.EnvoyContainer
+import pl.allegro.tech.servicemesh.envoycontrol.config.envoycontrol.EnvoyControlRunnerTestApp
+
+@SuppressWarnings("LargeClass")
+internal class IncomingPermissionsParamsInEndpointPathTest : EnvoyControlTestConfiguration() {
+
+    companion object {
```

Comment:
Shouldn't we start writing tests in a new way presented by @pbetkier https://github.com/allegro/envoy-control/pull/143 ?

Reply:
+1, there is already one incoming permissions test in new format, maybe it will help: https://github.com/allegro/envoy-control/blob/629e882c1acc00e46487175aef31d82c838cf46b/envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/permissions/TlsClientCertRequiredTest.kt